Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1474,6 +1474,13 @@ void FrameLoader::loadURL(FrameLoadRequest&& frameLoadRequest, const String& ref
     if (m_inStopAllLoaders || m_inClearProvisionalLoadForPolicyCheck)
         return;
 
+#if PLATFORM(WPE) || PLATFORM(GTK)
+    if ((m_policyDocumentLoader && m_policyDocumentLoader->isRequestFromClientOrUserInput()) || (m_provisionalDocumentLoader && m_provisionalDocumentLoader->isRequestFromClientOrUserInput())) {
```

Comment:
Why don't you have to check `m_documentLoader` as well here?

Reply:
If m_policyDocumentLoader or m_provisionalDocumentLoader exist, then there is ongoing initial loading process. m_documentLoader is always valid(if I corretly understand the code) and it is set in transitionToCommitted.